dali/sasha/saruncmd.cpp

```diff
@@ -46,6 +47,61 @@ class CSashaCmdExecutor : public CInterfaceOf<ISashaCmdExecutor>
         cmd->getId(0, lastServerMessage);
         return true;
     }
+    bool listWorkunit(ListWURequests *req, bool dfu) const
+    {
+        lastServerMessage.clear();
+        Owned<ISashaCommand> cmd = createSashaCommand();
+        cmd->setAction(req->includeDT ? SCA_LISTDT : SCA_LIST);
+        if (dfu)
+            cmd->setDFU(true);
+        cmd->setArchived(req->archived);
+        cmd->setOnline(req->online);
+        if (!req->outputFields.isEmpty())
+            cmd->setOutputFormat(req->outputFields);
+
+        cmd->addId(req->wuid.isEmpty() ? "*" : req->wuid.get());
+        if (!req->cluster.isEmpty())
+            cmd->setCluster(req->cluster.get());
+        if (!req->owner.isEmpty())
+            cmd->setOwner(req->owner.get());
+        if (!req->jobName.isEmpty())
+            cmd->setJobName(req->jobName.get());
+        if (!req->state.isEmpty())
+            cmd->setState(req->state.get());
+
+        if (!req->fromDate.isEmpty())
+            cmd->setAfter(req->fromDate.str()); //In CArchivedWUReader, the After is used as fromDate.
+        if (!req->toDate.isEmpty())
+            cmd->setBefore(req->toDate.str()); //In CArchivedWUReader, the Before is used as toDate.
+
+        cmd->setStart(req->pageStartFrom); //In CArchivedWUReader, the Start is the index of the first item to be returned.
+        cmd->setLimit(req->pageStartFrom + req->pageSize); //In CArchivedWUReader, the Limit is the index after the last item to be returned.
```

Comment:
see comment in saarch.cpp.
At the moment, I don't think a pageStartFrom should be part of the request, but rather paging should use the previous approach of specifying the beforeWu/afterWu.
And the count/limit should be an absolute number, i.e. not offset+count.